enhancements/sdk-external-and-pluggable-validations.md

```diff
@@ -0,0 +1,376 @@
+python file executable or supply a shell script to be invoked.
+
+##### Validator results
+
+As stated earlier, each validator should be some executable that accepts a
+single argument, the bundle root directory. Th validators should also output
+ManifestResult JSON to stdout.
+
+Because the [existing validators](#default-validators-run-by-operator-sdk)
+currently return a [ManifestResult][manifest-result], it seems logical that we
+use the same object as JSON for the output.
+
+The validator executable should exit non-zero ONLY if the entrypoint failed to
+run NOT if the bundle validation fails.
+
+For example, let's say the validator is given a path to the gatekeeper bundle.
+The validator should validate the given bundle and output the ManifestResult JSON.
+Here is an example of a run:
+
+```json
+{
+    "Name": "gatekeeper-operator.v0.2.0-rc.3",
+    "Errors": null,
+    "Warnings": [
+        {
+            "Type": "CSVFileNotValid",
+            "Level": "Warning",
+            "Field": "",
+            "BadValue": "",
+            "Detail": "(gatekeeper-operator.v0.2.0-rc.3) csv.Spec.minKubeVersion is not informed. It is recommended you provide this information. Otherwise, it would mean that your operator project can be distributed and installed in any cluster version available, which is not necessarily the case for all projects."
+        }
+    ]
+}
+```
+
+The above JSON will be read by the `operator-sdk` during `bundle validate`
+command and output the results as it does today. The example below shows what
+`operator-sdk bundle validate` would printout if given the ManifestResult from
+above.
+
+```json
+{
+    "passed": true,
+    "outputs": [
+        {
+            "type": "warning",
+            "message": "Warning: Value : (gatekeeper-operator.v0.2.0-rc.3) csv.Spec.minKubeVersion is not informed. It is recommended you provide this information. Otherwise, it would mean that your operator project can be distributed and installed in any cluster version available, which is not necessarily the case for all projects."
+        }
+    ]
+}
+```
+
+Allowing the validators to output ManifestResult should make it easier to
+transition existing validators to the external format with minimal code.
+
+##### Migrating existing validator to executable
+
+In the short to near term, you create a new `main.go` for each of the validators.
+Then import the validator code from operator-framework/api. The `main.go` would
+take in one (1) argument, the bundle directory.
+
+Since the [existing validators](#default-validators-run-by-operator-sdk) already
+output `ManifestResult`, it's easiest if we simply print that out as JSON to
+stdout.
+
+An example POC that takes an existing validator and outputs `ManifestResult` can
+be found at [validator-poc][validator-poc1]
+
+Another example of a migration can be find at [ocp-olm-catalog-validator][camila-poc].
+This particular example does NOT yet output `ManifestResult` format.
```

Comment:
We have no 2 external validators that can be used to be tested checked with SDK: 
- https://github.com/redhat-openshift-ecosystem/ocp-olm-catalog-validator
- https://github.com/k8s-operatorhub/bundle-validator

Reply:
I will update the examples.

Reply:
@camilamacedo86 so did we make these to be used by the external validator feature? or they were stand-alone ones?